webapp/channels/src/actions/views/channel.ts

```diff
@@ -306,17 +307,17 @@ export function loadPostsAround(channelId: string, focusedPostId: string) {
         dispatch({
             type: ActionTypes.INCREASE_POST_VISIBILITY,
             data: channelId,
-            amount: data.order.length,
+            amount: data!.order.length,
         });
         return {
-            atLatestMessage: data.next_post_id === '',
-            atOldestmessage: data.prev_post_id === '',
+            atLatestMessage: data!.next_post_id === '',
```

Comment:
I see several `!` in this file. How safe are they?

Reply:
Generally, they're as safe as they were before. 😅

Before, TS didn't check the type since data is an `any`, and now we're just telling it to trust us that these are defined. In a lot of these cases, we can know that `data` is defined because there's no `error` field, but because the current type for `ActionResult` is `{data?: T; error?: ServerError}`, TS doesn't know that we only define one or the other. I'd like to improve `ActionResult` still, but that's not done in any of these PRs